src/Voron/Data/BTrees/TreePage.cs

```diff
@@ -129,12 +129,19 @@ public ushort* KeysOffsets
                         high = position - 1;
                 }
 
-                if (lastMatch > 0) // found entry less than key
+                if (backward)
                 {
-                    position++; // move to the smallest entry larger than the key
+                    if (lastMatch < 0)  // found entry greater than key
+                        position--; // move to the largest entry smaller than the key
+                }
+                else
+                {
+                    if (lastMatch > 0) // found entry less than key
+                        position++; // move to the smallest entry larger than the key
+
+                    Debug.Assert(position < ushort.MaxValue);
```

Comment:
Can you please explain this assertion? Why do you validate position against `ushort.MaxValue`? Shouldn't it be done against  `NumberOfEntries`?

Reply:
This code was already there, I just moved it to the `else` scope